compiler/rustc_data_structures/src/sip128.rs

```diff
@@ -1,21 +1,31 @@
 //! This is a copy of `core::hash::sip` adapted to providing 128 bit hashes.
 
-use std::cmp;
 use std::hash::Hasher;
-use std::mem;
+use std::mem::{self, MaybeUninit};
 use std::ptr;
 
 #[cfg(test)]
 mod tests;
 
+const ELEM_SIZE: usize = mem::size_of::<u64>();
+const BUFFER_SIZE_ELEMS: usize = 8;
+const BUFFER_SIZE_BYTES: usize = BUFFER_SIZE_ELEMS * ELEM_SIZE;
+const BUFFER_SIZE_ELEMS_SPILL: usize = BUFFER_SIZE_ELEMS + 1;
+const BUFFER_SIZE_BYTES_SPILL: usize = BUFFER_SIZE_ELEMS_SPILL * ELEM_SIZE;
+const BUFFER_SPILL_INDEX: usize = BUFFER_SIZE_ELEMS_SPILL - 1;
```

Comment:
I can't resist bikeshedding the names... `size` usually means "size in bytes" and `capacity` usually means "maximum number of elements". So I suggest `BUFFER_CAPACITY`, `BUFFER_SIZE`, `BUFFER_SPILL_CAPACITY`, `BUFFER_SPILL_SIZE` for the first four.